Two-body-scf-guess.py

```diff
@@ -0,0 +1,71 @@
+from wrap_pyscf import *
+import numpy as np
+
+
+mol = [("O", 0.216486, -1.989005, 0.035297),
+       ("H", -0.603911, -1.505274, -0.033552),
+       ("H", 0.138672, -2.535875, 0.799566),
+       ("H", 1.505276, -0.603920, 0.033505),
+       ("O", 1.989004, 0.216483, -0.035296),
+       ("H", 2.535922, 0.138690, -0.799534),
+       ("O", -1.989001, -0.216487, -0.035297),
+       ("H", -2.535947, -0.138688, -0.799513),
+       ("H", -1.505273, 0.603920, 0.033493),
+       ("O", -0.216485, 1.989007, 0.035294),
+       ("H", -0.138694, 2.535879, 0.799565),
+       ("H", 0.603919, 1.505277, -0.033513)] 
+
+
+def two_body_guess(monomers, dimers, bs):
+    y = len(monomers)
+    for dimers_index in dimers:
+        dimers_index = dimers.index(dimers_index)
+        dimers_index = dimers[dimers_index]
+        for dimer in dimers_index:
+            dimer = dimers_index.index(dimer)
+            dimer = dimers_index[dimer]
+            dimer_dens = density(dimer, bs)
+        if dimers_index == dimers[0]:
+            for dimers_index in dimers:
+                a = sum(dimers[0])
+                rho[0:m, 0:m] = a[0:m, 0:m]-(y-2)*density(monomer[0], bs)
+                for dimer in dimers_index:
+                    x = dimers_index.index(dimer)
+                    rho[0:m, x*m+m:x*m+2*m] = dimer_dens[0:m, m:2*m]
+                    rho[x*m+m:x*m+2*m, 0:m] = dimer_dens[m:2*m, 0:m]
+        if dimers_index == dimers[y-1]:
+            for dimers_index in dimers:
+                b = dimers.index(dimers_index)
+                c = sum(dimers_index)
+                rho[b*m:(b+1)*m, b*m:(b+1)*m] = c[m:2*m, m:2*m]-(y-2)*density(monomer[b], bs)
+        if dimers_index != dimers[0]:
+            if dimers_index != dimers[y-1]:
+                for dimers_index in dimers:
+                    b = dimers.index(dimers_index)
+                    x = dimers_index.index(dimer)
+                    list1 = []
+                    for dimer in dimers_index:
+                        if x < b:
+                            list2 = []
+                            list2.append(dimer_dens)
+                            d = sum(list2)
+                            list1.append(d[m:m*2, m:m*2]) 
+                        if x >= b:
+                            list3 = []
+                            list3.append(dimer_dens)
+                            e = sum(list3)
+                            list1.append(e[0:m, 0:m])
+                            rho[b*m:(b+1)*m, x*m+m:x*m+2*m] = dimer_dens[0:m, m:2*m]
+                            rho[x*m+m:x*m+2*m, b*m:(b+1)*m] = dimer_dens[m:2*m, 0:m]
+                        f = sum(list1)
+                        rho[b*m:(b+1)*m, b*m:(b+1)*m] = f-(y-2)*density(monomer[b], bs)
+    nitr = run_scf(mol, bs, rho)
+    print(a)
+          
+monomers = [mol[0:3], mol[3:6], mol[6:9], mol[9:12]]
+# monomers = [A, B, C, D]
+dimers = [[mol[0:6], (mol[0:3]+mol[6:9]), (mol[0:3]+mol[9:12])], [mol[0:6], mol[3:9], (mol[3:6]+mol[9:12])], 
```

Comment:
Inside dimers[ ] (with index b) there is a list that contains all dimers (each dimer has index x) made from each monomer, in the order set by the monomers. One list, in dimers[ ], has dimers made with A, the others with B, so on. For monomer M I sum the densities in each list for rho's MM diagonal densities. 

There are repeat dimers because I use b index and x index to build rho. If you want I can explain my code over a zoom call, it would be easier to explain. After I can implement your suggestions. Let me know what yoy want to do.

Reply:
I think I see what you're doing now. Assuming I understand correctly, you want to run the monomer indices `b` and `x` over unrestricted ranges (i.e. all monomers) versus I had envisioned it as `b` unrestricted, `x` restricted to greater than equal to `b`. If you want to do it your way that's fine for now. I'd still recommend refactoring the code into subroutines though.